src/lib/swapper/api.ts

```diff
@@ -127,9 +133,9 @@ export type GetTradeQuoteInput =
   | GetEvmTradeQuoteInput
   | GetCosmosSdkTradeQuoteInput
 
-export type BuildTradeInput = RequireFields<GetTradeQuoteInput, 'receiveAddress'> & {
+export type BuildTradeInput = GetTradeQuoteInput & {
+  wallet?: HDWallet
```

Comment:
Let's keep this one required for now and remove it entirely as a follow-up, doing the same "get address from wallet, eip checks yadi yada" dance as a wrapper